### PR TITLE
Merge branch 'branch-24.02' into main

### PR DESCRIPTION
**Pull request by NvTimLiu**
Merge branch 'branch-24.02' into main

Change version to 24.02.0

Note: please merge this change with **"Create a merge commit"**

### CODE REVIEW

**Comment by NvTimLiu** [comment]
This PR is not the final merge to main change.

Draft the PR for the in-advance review purpose

NOTE: There are still some open PRs targeting 24.02, will do the final merge after they get merged.

https://github.com/NVIDIA/spark-rapids/pulls?q=is%3Apr+is%3Aopen+base%3Abranch-24.02+

File: CHANGELOG.md [hunk 1]
```diff
@@ -125,7 +125,7 @@ Generated on 2024-01-31
 |[#10123](https://github.com/NVIDIA/spark-rapids/pull/10123)|Change version to v23.12.1 [skip ci]|
 |[#10122](https://github.com/NVIDIA/spark-rapids/pull/10122)|Init changelog for v23.12.1 [skip ci]|
 |[#10121](https://github.com/NVIDIA/spark-rapids/pull/10121)|[DOC] update download page for db hot fix  [skip ci]|
-|[#10116](https://github.com/NVIDIA/spark-rapids/pull/10116)|Upgrade to 23.12.1|
+|[#10116](https://github.com/NVIDIA/spark-rapids/pull/10116)|Upgrade to 23.12.1-SNPASHOT|
```
Comment (jlowe) [inline CHANGELOG.md:128]: Typo that's not in branch-24.02?  Curious how this change happened since it should just be a merge.

Comment (NvTimLiu) [reply CHANGELOG.md:128]: Good catch! thanks @jlowe 

I found the root cause, that's my fault:

On dev branch-24.02 it is `23.12.1-SNAPSHOT` :  https://github.com/NVIDIA/spark-rapids/blob/branch-24.02/CHANGELOG.md


But, on main branch, I replaced all `23.12.1-SNAPSHOT` to `23.12.1`, where I should NOT modify CHANGELOG.md file :
https://github.com/NVIDIA/spark-rapids/commit/e83c1c1fdd70568cdbdb3ff13d43f7e5f6cb8ae5


So we should keep `23.12.1-SNPASHOT` here, to make it the same as what dev branch is.

**Comment by NvTimLiu** [comment]
Oh I should create a PR targeting to branch-24.02 for : https://github.com/NVIDIA/spark-rapids/pull/10383/commits/19cd490a6ea06b9639bc1ae2bd191351c5e3170d

Let me revert it from `merge-to-main`

**Comment by NvTimLiu** [comment]
Shall we update `-2024` copyright targeting to main, as we change version to `24.02.0` from `24.02.0-SNAPSHOT` branch-24.02(on which copyright is up to 2.23)? 

e.g.  shuffle-plugin/pom.xml on `branch-24.02` :  https://github.com/NVIDIA/spark-rapids/blob/branch-24.02/shuffle-plugin/pom.xml#L3

shuffle-plugin/pom.xml changed`merge-to-main` :  https://github.com/NVIDIA/spark-rapids/pull/10383/commits/f56520c87321551c37ce532f8072acdf7ba4d42c#diff-c29b084f7d961795298e283df6dcc8b6a5a549abe6d1153347faa414e932acdb

File: CHANGELOG.md (hunk 1 above)
Comment (jlowe) [reply CHANGELOG.md:128]: The dev branch does not have 23.12.1-SNPASHOT, it has 23.12.1-SNAPSHOT.  I'm curious how SNAPSHOT became SNPASHOT.

**Comment by jlowe** [comment]
> Shall we update -2024 copyright targeting to main, as we change version to 24.02.0 from 24.02.0-SNAPSHOT

Yes.  If we are not simply merging but also modifying with this PR, any files modified instead of simply merged need to have updated copyrights.

File: CHANGELOG.md (hunk 1 above)
Comment (NvTimLiu) [reply CHANGELOG.md:128]: Aha, sorry, this is a typo when I fixed merge conflicts to main from dev branch, let me fix it

**Comment by NvTimLiu** [comment]
This commit history of merge-to-main got messed up with:
 
[19cd490](https://github.com/NVIDIA/spark-rapids/commit/19cd490a6ea06b9639bc1ae2bd191351c5e3170d)

[7610623](https://github.com/NVIDIA/spark-rapids/pull/10383/commits/76106238d2953b3ab70c6c8ca568e63ce19f3e94)

[735a06f](https://github.com/NVIDIA/spark-rapids/pull/10383/commits/735a06f8c71cdb0e76aadb51fb5fe092a7bf45bc)

and fix typo


I'd like to close this one, and create a new `merge-to-main` PR instead.

**Comment by NvTimLiu** [comment]
Close this PR, it's replaced by https://github.com/NVIDIA/spark-rapids/pull/10395

**Pull request closed without merging**